### DISCUSSION TITLE
Wrong arc start and end points, center point

### DISCUSSION THREAD

**Discussion by gaoxipeng** [mozman/ezdxf#871] [Q&A]
Hello, In the dxf file of this attachment, it is incorrect for the center of the arc to be the x coordinate, which should be 3900 in AutoCAD，start_ angle and end_ angle is also wrong
I tried to delete this arc and draw a shape similar to it, and got the correct result, but now this arc is incorrect.

ARC center: (-3900.0, 12242.500000001, 0.0), radius: 877.5000000000002, start_angle: 90.0, end_angle: 178.5307164186462
Start point: (3900.0, 13120.000000001, 0.0), end point: (4777.211491032807, 12265.000000001, 0.0)

[Drawing1_t3.zip](https://github.com/mozman/ezdxf/files/11330722/Drawing1_t3.zip)

**Comment by mozman** [comment]
The values are correct:

![image](https://user-images.githubusercontent.com/4230966/234553873-b4a0774a-4487-4169-84cb-53a262fb6e33.png)

The [OCS](https://ezdxf.mozman.at/docs/concepts/ocs.html) is defined by the extrusion vector, which is (0, 0,-1) in this case:

![image](https://user-images.githubusercontent.com/4230966/234554040-7480a36a-f29f-4837-9d8b-2eeb33f49928.png)

The WCS is only equal to the OCS when the extrusion vector is (0, 0, 1)  the WCS z-axis.

The [upright](https://ezdxf.mozman.at/docs/upright.html) module can help to convert an inverted extrusion vector to (0, 0, 1).
